### PR TITLE
added mxnet.js tests

### PR DESCRIPTION
**Pull request by arank**
Added docker build for emcc from src to compile mxnet.js on ubuntu

### CODE REVIEW

File: tests/ci_build/Dockerfile.emscripten [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu
+MAINTAINER Aran Khanna <arankhan@amazon.com>
```
Comment (mli) [inline tests/ci_build/Dockerfile.emscripten]: MAINTAINER is deprecated https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

File: tests/ci_build/Dockerfile.emscripten [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+RUN apt-get update && apt-get -y upgrade
+
+# TOOLCHAIN DEPS
+RUN apt-get install -y python python-setuptools python-pip python-dev unzip gfortran
+RUN apt-get install -y git nodejs build-essential cmake
+
+# BUILD EMSCRIPTEN
+RUN git clone https://github.com/kripken/emscripten.git
+RUN git clone https://github.com/kripken/emscripten-fastcomp
+RUN cd emscripten-fastcomp && git clone https://github.com/kripken/emscripten-fastcomp-clang tools/clang && mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;JSBackend" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DCLANG_INCLUDE_EXAMPLES=OFF -DCLANG_INCLUDE_TESTS=OFF && make
```
Comment (mli) [inline tests/ci_build/Dockerfile.emscripten]: use \ to split it into multiple lines for better readability

**Comment by arank** [comment]
This docker is used to test the build process for MXNet.js

**Comment by arank** [comment]
Just updated the file. This will be used to test the build process of mxnet.js from the latest emscripten version (built from source) to ensure stability. Emscripten has to be built on ubuntu (and windows) and not apt installed since the apt version on ubuntu is old and fails when building MXNet.js, however on Mac the brew version is sufficiently recent such that this isn't a problem. If we want to standardize on a particular versions of emscripten we can checkout specific hashes in the repo before build.

**Pull request merged**